src/content/developers/docs/oracles/index.md

```diff
@@ -1,314 +1,86 @@
 
-Chainlink VRF (Verifiable Random Function) is a provably-fair and verifiable source of randomness designed for smart contracts. Smart contract developers can use Chainlink VRF as a tamper-proof random number generation (RNG) to build reliable smart contracts for any applications which rely on unpredictable outcomes:
+Oracles act as a “bridge” connecting smart contracts and blockchains to the real world. Without oracles, smart contract applications would be limited in usefulness and real-world value. A blockchain oracle provides a mechanism for extracting real-world information which smart contracts can use to trigger state changes in the [Ethereum Virtual Machine](/developers/docs/evm/)
 
-- Blockchain games and NFTs
-- Random assignment of duties and resources (e.g. randomly assigning judges to cases)
-- Choosing a representative sample for consensus mechanisms
+Oracles differ based on the source of data (one or multiple sources), direction of information (inbound or outbound), trust models (centralized or decentralized) and system architecture (immediate-read, publish-subscribe, and request-response). We can also distinguish between oracles that only retrieve external data (input oracles) and ones that perform computation off-chain (computational oracles).
 
-Random numbers are difficult because blockchains are deterministic.
+## Why do smart contracts need oracles? {#why-do-smart-contracts-need-oracles}
 
-Working with Chainlink Oracles outside of data feeds follows the [request and receive cycle](https://docs.chain.link/docs/architecture-request-model) of working with Chainlink. They use the LINK token to send oracle providers oracle gas for returning responses. The LINK token is specifically designed to work with oracles and are based on the upgraded ERC-677 token, which is backwards compatible with [ERC-20](/developers/docs/standards/tokens/erc-20/).
-The following code, if deployed on the Kovan testnet will retrieve a cryptographically proven random number. To make the request, fund the contract with some testnet LINK token that you can get from the [Kovan LINK Faucet](https://kovan.chain.link/).
+Most developers see smart contracts as simply pieces of code running at specific addresses on the blockchain. However, a more [general view art contracts](/smart-contracts/) is that they are self-executing software programs capable of enforcing agreements between parties once specific conditions are met—which explains the term, “smart contracts.”
 
-```javascript
+But there’s a problem with using smart contracts to enforce agreements between people: blockchains are designed to be “deterministic”.  A [deterministic system](https://en.wikipedia.org/wiki/Deterministic_algorithm) is one that always produces the same results given an initial state and a particular input—there is no randomness or variation in the process of computing outputs from inputs. 
```

Comment:
The start of this paragraph could do with a rethink. It sounds like determinism in the code execution is a bug and oracles are a workaround. That's not the case - oracles are just bridges that enable real world data to influence on-chain events.

Reply:
Changed to: "But using smart contracts to enforce agreements between people isn't exactly straightforward, given that blockchains are designed to be “deterministic”. "